### PR TITLE
SDIT-705: ⚡️ Add in new prisons endpoint and deprecate old ones

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/AgencyResource_prisonsIntTest.kt [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      webTestClient.get()
+        .uri("/api/agencies/prisons")
+        .headers(setAuthorisation(listOf("ROLE_MAINTAIN_HEALTH_PROBLEMS")))
+        .header("Content-Type", MediaType.APPLICATION_JSON_VALUE)
+        .accept(MediaType.APPLICATION_JSON)
+        .exchange()
+        .expectStatus().isOk
+        .expectBody()
+        .jsonPath("\$.length()").value<Int> { assertThat(it).isGreaterThan(2) }
+        .jsonPath("\$[*].agencyId").value<List<String>> { assertThat(it).contains("BMI", "BXI", "TRO") }
```
Comment (andymarke) [inline src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/AgencyResource_prisonsIntTest.kt:41]: cut and paste error?

**Review by andymarke — DISMISSED** [review]
LGTM

File: src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/AgencyResource_prisonsIntTest.kt (hunk 1 above)
Comment (petergphillips) [reply src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/AgencyResource_prisonsIntTest.kt:41]: yup, now fixed

**Review by andymarke — APPROVED** [review]
lgtm

**Pull request merged**